### DISCUSSION TITLE
Trailing Slash without redirect

### DISCUSSION THREAD

**Discussion by shaije** [vercel/next.js#15155] [Help]
Hi, so i upgraded to latest canary version to explore the trailing slash fix in ssr.
i saw that now it redirects from /path/ to /path

is there a way to keep the trailing slash without this redirect?
i fear that this redirect will affect the performance, for example lighthouse do warn about useless redirects.

Thank you:)

**Comment by Janpot** [comment]
At the moment there is no way to disable this behavior. To avoid unnecessary redirects, links are automatically rewritten to conform to the `trailingSlash` setting. It's also using permanent redirects, these are cacheable so they should avoid a redirect when revisiting a link. Are you seeing different behavior?

**Reply by shaije** [reply]
Thank you for your reply.

This is the behavior i see also.

Our site is a lot of landing pages (hundred of thousand of pages..) of one time users so the concept of cache pages in the browser is not so important to us. 
What important to us is the speed and performance. for the last two years we used SSG across all site with next export, but now we have decided to move part of the site to SSR. the trailing slash is important for us because there are a lot of links of our site out there in the web with this trailing slash so we have to support it. 
So for what i see our last resort is to do custom server and support it, but then we will lose the advantage of Automatic Static Optimization to part of the pages.
Do you have maybe another idea for us?

Are you guys planning to support this behavior maybe in the soon future? the trailing slash without redirect? 

Thank you for your answer

**Reply by shaije** [reply]
@Janpot 
Thx!
